internal/repository/file.go

```diff
@@ -210,34 +187,39 @@ func (r *FileRepository) Parent(u fyne.URI) (fyne.URI, error) {
 //
 // Since: 2.0
 func (r *FileRepository) Child(u fyne.URI, component string) (fyne.URI, error) {
-	newURI := u.Scheme() + "://" + u.Authority()
-	newURI += path.Join(u.Path(), component)
+	child := strings.Builder{}
+
+	child.WriteString(u.Scheme())
+	child.WriteString("://")
+	child.WriteString(u.Authority())
+	child.WriteString(path.Join(u.Path(), component))
 
 	// stick the query and fragment back on the end
 	if query := u.Query(); len(query) > 0 {
-		newURI += "?" + query
+		child.WriteByte('?')
+		child.WriteString(query)
 	}
 	if fragment := u.Fragment(); len(fragment) > 0 {
-		newURI += "#" + fragment
+		child.WriteByte('#')
+		child.WriteString(fragment)
 	}
 
-	return storage.ParseURI(newURI)
+	return storage.ParseURI(child.String())
```

Comment:
Parent() strips query and fragment parts. If we can do so for Child() also, we can implement this simply as `return storage.NewFileURI(path.Join(u.Path(), component))`. Any thoughts @charlesdaniels ?